dnf5/commands/download/download.cpp

```diff
@@ -146,20 +156,31 @@ void DownloadCommand::run() {
         }
     }
 
-    if (!download_pkgs.empty()) {
-        libdnf5::repo::PackageDownloader downloader(ctx.base);
-
-        // for download command, we don't want to mark the packages for removal
-        downloader.force_keep_packages(true);
+    if (download_pkgs.empty()) {
+        return;
+    }
 
-        for (auto & [nevra, pkg] : download_pkgs) {
-            downloader.add(pkg);
+    if (url_option->get_value()) {
+        for (auto & [nerva, pkg] : download_pkgs) {
+            auto urls = pkg.get_remote_locations();
+            libdnf_assert(urls.empty(), "Failed to get mirror for package: \"{}\"", pkg.get_name());
```

Comment:
Oh, the condition has to be reversed: `!urls.empty()`
This way it always fails.

Reply:
No problem on the squash. I'm confused on the url.empty part. Wouldn't we want it to trigger an assert if the command when the vector is emptyl? Thank you